### ISSUE TITLE
Be able to access the event_type that triggered an event on the on_event automation.

### ISSUE THREAD

**Issue by ellull** [esphome/feature-requests#2743]
**Describe the problem you have/What new integration you would like**

The on_event automation of the event component can't use the event_type that triggered the event to apply some logic ("if" conditions for example) or just forward it to some other component.

So when an event is triggered, I would like to be able to access the event_type used to trigger the event on the on_event automation as I can access the new state in the binary_sensor.on_state automation (https://esphome.io/components/binary_sensor/#on-state) or the new published value on the text_sensor.on_value automation (https://esphome.io/components/text_sensor/#text-sensor-on-value).

**Please describe your use case for this integration and alternatives you've tried:**

My use case is that I want to perform different actions on the on_event automation depending on the event_type that triggered the event.

**Additional context**

None

**Comment by ellull** [comment]
My bad. I assumed I coudn't because I tried using the **_x_** variable (which is the convention in other automations) like:
```
    on_event:
      then:
        text_sensor.template.publish:
          id: triggered_event
          state: !lambda return x;
```
And the compilation gave an error that the _x_ variable didn't exist in the scope.
Also, the documentation does not mention it.

But after looking at the code I've found out that you already can access the event_type by using the **_event_type_** variable like:
```
    on_event:
      then:
        text_sensor.template.publish:
          id: triggered_event
          state: !lambda return event_type;
```

I will make the modification on the esphome-docs project to document this feature because I think it could be useful for many other people.
